theano/tensor/elemwise.py

```diff
@@ -696,7 +696,7 @@ def as_scalar(t):
 
             scalar_inputs = list(map(as_scalar, inputs))
             scalar_ograds = list(map(as_scalar, ograds))
-            scalar_igrads = self.scalar_op.grad(scalar_inputs, scalar_ograds)
+            scalar_igrads = self.scalar_op.L_op(scalar_inputs, outputs, scalar_ograds)
```

Comment:
I think on line 717, you need to add:

    if r in scalar_outputs:
        return outputs[scalar_outputs.index(r)]

This should do the good mapping between the scalar and tensor variable.